dapp/src/utils/hooks.js

```diff
@@ -1,57 +1,103 @@
 import { useEffect, useState, useRef } from 'react'
 import { useWeb3React } from '@web3-react/core'
-
-import { injected, connectorsByName, getConnectorImage } from './connectors'
+import { SafeAppConnector } from '@gnosis.pm/safe-apps-web3-react'
+
+import {
+  injected,
+  connectorsByName,
+  getConnector,
+  getConnectorImage,
+} from './connectors'
 import AccountStore from 'stores/AccountStore'
 import analytics from 'utils/analytics'
 
 export function useEagerConnect() {
   const { activate, active } = useWeb3React()
 
-  const [tried, setTried] = useState(false)
+  const [triedInjected, setTriedInjected] = useState(false)
+  const [triedSafeMultisig, setTriedSafeMultisig] = useState(false)
+  const [safeMultisigConnector, setSafeMultisigConnector] = useState(null)
+  const [connector, setConnector] = useState(null)
+  const [connectorIcon, setConnectorIcon] = useState(null)
+
+  // Instantiate Gnosis Safe web3-react connector
+  useEffect(() => {
+    // Requires global.window to be available to must be instantiated inside
+    // useEffect for Next.js to work
+    setSafeMultisigConnector(new SafeAppConnector())
+  }, []) // Do this on mount
```

Comment:
I think these to `useEffects` could be joined into one to reduce complexity with smth like this: 
```
useEffect(() => {
  if (!process.browser) return 

  const SafeMultisigConnector = new SafeAppConnector()
  ...

}, [process.browser])
```

Reply:
Done